src/pages/billing/Project.js

```diff
@@ -525,14 +525,22 @@ const ProjectDetail = ({ authorizeAndLoadAccounts, billingAccounts, billingProje
     { ms: 5000 }
   )
 
-  const { displayName = null } = _.find({ accountName: billingProject.billingAccount }, billingAccounts) || { displayName: 'No Access' }
+  // (CA-1586) For some reason the api sometimes returns string null, and sometimes returns no field, and sometimes returns null. This is just to be complete.
+  const billingProjectHasBillingAccount = !(billingProject.billingAccount === 'null' || _.isNil(billingProject.billingAccount))
+  const billingAccount = billingProjectHasBillingAccount ? _.find({ accountName: billingProject.billingAccount }, billingAccounts) : undefined
+
+  const billingAccountDisplayText = Utils.cond(
+    [!billingProjectHasBillingAccount, () => 'No linked billing account'],
+    [!billingAccount, () => 'No access to linked billing account'],
+    () => billingAccount.displayName || billingAccount.accountName
```

Comment:
I don't know if it's possible that a billing account has no display name, but this handles it in case it is.